### PR TITLE
base64 encode secrets in config_dump

### PR DESCRIPTION
**Pull request by GregHanson**
existing output includes `\n` chars.  This change makes it easier to pipe into openssl commands

### CODE REVIEW

**Comment by GregHanson** [comment]
existing output:
```
...
  "certificates": [
    {
      "identity": "spiffe://cluster.local/ns/default/sa/httpbin",
      "state": "Available",
      "ca_cert": [
        {
          "pem": "-----BEGIN CERTIFICATE-----\nMIIDmjCCAYKgAwIBAgIQRC35VXM230xACoqQ48BiBjANBgkqhkiG9w0BAQsFADA9\nMQ4wDAYDVQQKDAVJc3RpbzEYMBYGA1UEAwwPSW50ZXJtZWRpYXRlIENBMREwDwYD\nVQQHDAhjbHVzdGVyMTAeFw0yMzExMzAxNDMxNDdaFw0yMzExMzAxNDQzNDdaMAAw\nWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAS3q2C7pN+eKmtkaC31xZN6RCulFP6u\nrQfy18Em/95RnLKCSta9RVBq0Wk/PXpxCaA/FduEHewdrnt/ALwDi25mo4GdMIGa\nMA4GA1UdDwEB/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIw\nDAYDVR0TAQH/BAIwADAfBgNVHSMEGDAWgBTGiq5JaXmd3y8pkKMtPPOpPR+eejA6\nBgNVHREBAf8EMDAuhixzcGlmZmU6Ly9jbHVzdGVyLmxvY2FsL25zL2RlZmF1bHQv\nc2EvaHR0cGJpbjANBgkqhkiG9w0BAQsFAAOCAgEAZ951/4uCglshBoR9NcDcDTnD\ni7xr2HBexo1TQ8eltndO1f17k+admgMWY1PMKTwY3W0dvnuThv22Pb+8ssATtrO5\nFuf+KUVP52U1BTs9hJdaX6pnsgnIJVD2Ou6RG7e1fIr+93fQfOZ+2La0Bq61RlT+\n97RRMI5hfLuNl4X3EWeq7Vs1ef4YwAv1Fx9xPQ0UrrwwY+wn3bAD3q4h8++DsD8N\nm0oeUI0e8fhVQMsfMbTZNiv12Snrksj0xd/0cSwSc5juv8n5gqbhZw5QeY9xH8QB\nzMeXjLJ1XISD5pIxtEf5S4YToxj+h/lY2zCqOznI55EG2iyALh2SErqT/zIRYXmA\nF8Z+/BBfQzIub6OqMXDjGsCCFV6VFWuqnFZr6OaZlbzk0jEAx+ZTtzzQFgGQPbDZ\nS47NJPyC0WNcMybjzClWWuHYxw5h1aaECb+gFHZvKNHXvvAxKpxDIzBbiLfTDqfz\n48WEOz9EGPylJVs0fcLXsf+nApnYBnF/65xFmbAM9HLNQAC0HL0tUM7am3kL5RBR\nVl2Et52ZXIgPcLxks7WjUuD4tRSibWp7IfHXUuiQY/nPJCD55Xz9691b2KuqWZ0S\nGuIDmONnDZawJ9Yt3hKOZKnM3xjcwJO4R2Qvf7EFc9MnseeTxbE7d1VBQDQsbSB7\nvHoYHmrvM0XMR4QZEyQ=\n-----END CERTIFICATE-----\n",
          "serial_number": "90626214162050582197153766285372318214",
          "valid_from": "2023-11-30T14:31:47Z",
          "expiration_time": "2023-11-30T14:43:47Z"
        }
      ],
...
```

proposed updated output:
```
...
  "certificates": [
    {
      "identity": "spiffe://cluster.local/ns/default/sa/httpbin",
      "state": "Available",
      "ca_cert": [
        {
          "pem": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNkVENDQVYyZ0F3SUJBZ0lRWExTaHBoVjROblJMdlQ4ZzRONlFrekFOQmdrcWhraUc5dzBCQVFzRkFEQVkKTVJZd0ZBWURWUVFLRXcxamJIVnpkR1Z5TG14dlkyRnNNQjRYRFRJek1URXpNREUxTVRreU1sb1hEVEl6TVRJdwpNVEUxTWpFeU1sb3dBREJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUFCTXJLVVdWWk5DTXZ3Rms4Cm9GOVFtanVGeG51LzhTVUxFZEU1Q0p4ZXZENmdNSEZvNVR4enRoRkFhTGRqb1FYMkZvejdQRFJFTThObnFxWGsKOEF2Wm9UdWpnWjB3Z1pvd0RnWURWUjBQQVFIL0JBUURBZ1dnTUIwR0ExVWRKUVFXTUJRR0NDc0dBUVVGQndNQgpCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFQWpBQU1COEdBMVVkSXdRWU1CYUFGSG5GcDB5RVpsRVM1Y2ZtCmd6eHlvcENCU2J3c01Eb0dBMVVkRVFFQi93UXdNQzZHTEhOd2FXWm1aVG92TDJOc2RYTjBaWEl1Ykc5allXd3YKYm5NdlpHVm1ZWFZzZEM5ellTOW9kSFJ3WW1sdU1BMEdDU3FHU0liM0RRRUJDd1VBQTRJQkFRQmsyejVadmhKdAo5ajdKampyT2NwOStlZk5KY2Y2WE8rYjkwWHdLR0RrYUp1SW9QOE5iT2tKRTdRVVg0UUsremNPakF2R1o2dGQ2ClZlRzBFSWdteUVuOW51cytCbXVCYm42cVFTMStOL3ROYUZNMlJZL0NqVzRlZ3lJWktvRlQ2T3d4aVAvTERLdVYKRnYzRTdqT0lISjNYR3ZGSGpUZlM2elRMcjI4TGJlM2dtdXRtR096L1BwRDB0K2YyMFZtODFUQSswNWZWMGlOcQpDRzRSY2pIVnVKcXdBeGJ0RHdJUzQ1YlJuR2JkMmcrZS9RM1RSZmhKc2RsU1RXaitnU2ptZWZuMWt2cEhaMlBPCjljTkhiVUFHN0hLVmdIT1lrdWxRSGRVOGZKcVQvWU12MWR2aU4wdVRKY2x5UjhqL2crV296cmxhM25uaVZ5UEMKNUhEWnN2TVdZd3AzCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K",
          "serial_number": "123226867673214208024776012110025887891",
          "valid_from": "2023-11-30T15:19:22Z",
          "expiration_time": "2023-12-01T15:21:22Z"
        }
      ],
...
```

File: src/admin.rs [hunk 1]
```diff
@@ -186,10 +187,7 @@ async fn handle_dashboard(_req: Request<Incoming>) -> Response<Full<Bytes>> {
 fn x509_to_pem(x509: &X509) -> String {
     match x509.to_pem() {
         Err(e) => format!("<pem construction error: {e}>"),
-        Ok(vec) => match String::from_utf8(vec) {
-            Err(e) => format!("<utf8 decode error: {e}>"),
-            Ok(s) => s,
-        },
+        Ok(vec) => base64::encode_block(&vec),
```
Comment (ilrudie) [inline src/admin.rs:190]: Envoy also encodes certs in base64 so this seams good.

Should the field name be changed to reflect that it's not a plaintext PEM anymore?

**Review by ilrudie — APPROVED** [review]
This looks reasonable to me

File: src/admin.rs (hunk 1 above)
Comment (ilrudie) [reply src/admin.rs:190]: inline_bytes I don't love either so I didn't want to specifically call out that name 😂 field makes sense including inline vs being a reference to a file though so maybe it's not all bad.

Comment (hzxuzhonghu) [reply src/admin.rs:190]: No need to change that, since base64 is also intuitive

**Review by hzxuzhonghu — APPROVED** [review]
LGTM.

**Pull request merged**